qa/L0_http/http_input_size_limit_test.py

```diff
@@ -0,0 +1,315 @@
+            np.array_equal(output_array, small_input),
+            "Response data does not match input data",
+        )
+
+    def test_default_limit_rejection_json(self):
+        """Test JSON inputs with default limit"""
+        model = "onnx_zero_1_float32"
+
+        # Test case 1: Input just over the limit (should fail)
+        shape_size = 2**24 + 32  # Just over 64MB of float32 data
+
+        payload = {
+            "inputs": [
+                {
+                    "name": "INPUT0",
+                    "datatype": "FP32",
+                    "shape": [1, shape_size],
+                    "data": [1.0] * shape_size,
+                }
+            ]
+        }
+
+        headers = {"Content-Type": "application/json"}
+        response = requests.post(
+            self._get_infer_url(model), headers=headers, json=payload
+        )
+
+        # Should fail with 400 bad request with default limit
+        self.assertEqual(
+            400,
+            response.status_code,
+            "Expected error code for oversized JSON request, got: {}".format(
+                response.status_code
+            ),
+        )
+
+        # Verify error message contains size limit info
+        error_msg = response.content.decode()
+        self.assertIn(
+            "exceeds the maximum allowed value",
+            error_msg,
+            "Expected error message about exceeding max input size",
+        )
+
+        # Test case 2: Input just under the limit (should succeed)
+        shape_size = 2**24 - 32  # Just under 64MB of float32 data
+
+        payload = {
+            "inputs": [
+                {
+                    "name": "INPUT0",
+                    "datatype": "FP32",
+                    "shape": [1, shape_size],
+                    "data": [1.0] * shape_size,
+                }
+            ]
+        }
+
+        response = requests.post(
+            self._get_infer_url(model), headers=headers, json=payload
+        )
+
+        # Should succeed with 200 OK
+        self.assertEqual(
+            200,
+            response.status_code,
+            "Expected success code for JSON request within size limit, got: {}".format(
+                response.status_code
+            ),
+        )
+
+        # Verify we got a valid response
+        result = response.json()
+        self.assertIn("outputs", result, "Response missing outputs field")
+        self.assertEqual(1, len(result["outputs"]), "Expected 1 output")
+        self.assertEqual(
+            shape_size,
+            result["outputs"][0]["shape"][1],
+            f"Expected shape {[1, shape_size]}, got {result['outputs'][0]['shape']}",
+        )
+
+    def test_large_input_raw_binary(self):
+        """Test raw binary input larger with custom limit set"""
+        model = "onnx_zero_1_float32"
+
+        # Test case 1: Input just over our configured limit (should fail)
+        large_input = np.ones(2**25 + 32, dtype=np.float32)  # Just over 128MB
```

Comment:
In `test.sh` it looks like this is set to `2**27` (128MB) but here in the pytest on this line, it looks like `2**25`=32MB + ~32bytes. 

Why is this failing?

Reply:
Are you actually seeing this "test case 1" fail when sending a payload of ~32MB when the limit is custom set to 128MB? Or am I misreading something?

Reply:
Oh - I think this is because its 32MB elements * 4bytes (fp32) each. Please update the comment to make this more clear.